src/components/field/svg/FieldGeneratedLines.tsx

```diff
@@ -41,8 +45,8 @@ class FieldPathLines extends Component<Props, State> {
           /
           {this.context.model.uiState.selectedPathGradient !=
             PathGradients.None.name &&
-            generated.length > 1 &&
-            generated.map((point, i, arr) => {
+            trajectory.length > 1 &&
+            trajectory.map((point, i, arr) => {
               if (i == arr.length - 1) {
                 return <></>;
               }
```

Comment:
reverse `if (!pathGradient) {` and `if (i == arr.length - 1) {`. technically, accessing the second point is more expensive than checking if something exists.

Reply:
we can exit much faster than acccessing something for no reason.

Reply:
ACTUALLY. we are already comparing if != PathGradients.None; therefore, implicitly checking against `undefined`, so it wouldn't change midway through map and before it (would invalidate current render...)